packages/rules-engine/src/lib/custom-rule-engine/helpers/cast-value.ts

```diff
@@ -0,0 +1,16 @@
+import { CustomRuleInput } from '@/lib/custom-rule-engine/types/custom-rule-input';
+
+export const castValue = (options: CustomRuleInput['options'], value: string) => {
+  switch (options.cast) {
+    case 'String':
+      return String(value);
+    case 'Number':
+      return Number(value);
+    case 'Boolean':
+      return Boolean(value);
+    case 'Date':
+      return new Date(value);
```

Comment:
Need to validate that the date (for example, `new Date('invalid').getTime()` will output `NaN`)